convert/src/label_studio_to_badgerdoc/label_studio_to_badgerdoc_use_case.py

```diff
@@ -0,0 +1,257 @@
+import tempfile
+from pathlib import Path
+from uuid import uuid4
+
+import requests
+from botocore.exceptions import ClientError
+from fastapi import HTTPException
+from tenant_dependency import TenantData
+
+from src.config import DEFAULT_PAGE_BORDER_OFFSET, settings
+from src.label_studio_to_badgerdoc.badgerdoc_format.annotation_converter import (
+    AnnotationConverter,
+)
+from src.label_studio_to_badgerdoc.badgerdoc_format.badgerdoc_format import (
+    BadgerdocFormat,
+)
+from src.label_studio_to_badgerdoc.badgerdoc_format.pdf_renderer import (
+    PDFRenderer,
+)
+from src.label_studio_to_badgerdoc.badgerdoc_format.plain_text_converter import (
+    TextToBadgerdocTokensConverter,
+)
+from src.label_studio_to_badgerdoc.models.label_studio_models import (
+    LabelStudioModel,
+    S3Path,
+)
+from src.logger import get_logger
+
+LOGGER = get_logger(__file__)
+
+
+class LabelStudioToBDConvertUseCase:
+    def __init__(
+        self,
+        s3_client,
+        current_tenant: str,
+        token_data: TenantData,
+        s3_input_annotation: S3Path,
+        s3_output_bucket: str,
+        s3_output_file_id: int,
+    ) -> None:
+        page_border_offset = DEFAULT_PAGE_BORDER_OFFSET
+        self.plain_text_converter = TextToBadgerdocTokensConverter(
+            page_border_offset=page_border_offset
+        )
+        self.pdf_renderer = PDFRenderer(page_border_offset=page_border_offset)
+        self.annotation_converter = AnnotationConverter()
+        self.badgerdoc_format = BadgerdocFormat()
+        self.s3_client = s3_client
+
+        self.current_tenant = current_tenant
+        self.token_data = token_data
+        self.request_headers = {
+            "X-Current-Tenant": self.current_tenant,
+            "Authorization": f"Bearer {self.token_data.token}",
+        }
+        self.s3_input_annotation = s3_input_annotation
+        self.s3_output_bucket = s3_output_bucket
+        self.s3_output_file_id = s3_output_file_id
+
+    def execute(self) -> None:
+        label_studio_format = self.download_label_studio_from_s3(
+            self.s3_input_annotation
+        )
+        LOGGER.debug(f"label studio format {label_studio_format=}")
```

Comment:
Make it in printf-like style `("foo %s", arg)`

Reply:
Thanks for the tip. I didn't know, that 'logging' is optimized for print-like style messages.
Reworked